WcaOnRails/app/webpacker/lib/wca-data.js.erb

```diff
@@ -146,6 +146,8 @@ export const delegateRegionsStatus = {
   senior_delegate: 'senior_delegate',
 };
 
+export const userRolesSortParams = <%= Api::V0::UserRolesController::SORT_PARAMS.each_with_object({}) { |(key, _), new_hash| new_hash[key] = key }.to_json %>;
```

Comment:
WAIT WHAT IN ALL HEAVENS NAME IS THIS??!?!???!?!!

Reply:
These are the sortParams used to sort userRoles. Didn't get a better name 😅 Do you have any idea?

Reply:
The name is fine. I am very, very confused by the "each_with_hash" hack. No idea what it does or why it's necessary, but I'm 99.97% sure it's overly complicated

Reply:
Reason why it's necessary: I can use it in React instead of directly using the string. I'm trying to create following JSON:
```
{
  startDate: startDate,
  status: status,
  ...
}
```

Yes, I agree it's a bit complex. I've made it a bit more clean now. Hope it's good now.

Reply:
There's an even simpler way:
```ruby
array_of_values.index_by { |v| doe_something_with_value(v) }
```

Reply:
Why does it need to have the shape `{ key: key }`?
Can't you just access `SORT_PARAMS.keys` and call it a day?

Reply:
But it cannot be access like a map right? Like we do `groupTypes.board`, etc.

Reply:
Yes, you cannot access a list of keys like a map.
But if your goal is to access it like a map, and the value is the same like the key you're accessing, then you already know what the result will be based on the access you're trying.

Reply:
If you really construct this like `const foo = { key: key }` then you already know that `foo.some_crazy_team` will definitely return `some_crazy_team`

Reply:
Okay, so hard-coding the values directly as a string in the javascript code is a better approach?

Reply:
Questioning why and where you need these values is a better approach.
Why does it have to be an object that has the same keys an values?